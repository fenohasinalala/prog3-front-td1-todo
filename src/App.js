import { useState } from "react";
import "./styles.css";
import List from "./components/List";
import { getMatches } from "./provider/getMatches";

export default function App() {
  const [toDos, setToDos] = useState([]);
  const [dones, setDones] = useState([]);

  const toDone = (id) => {
    const toDosCopy = [...toDos];
    const toDosUpdate = toDosCopy.filter((f) => f.id !== id);
    const toAdd = toDosCopy.filter((f) => f.id === id);
    setToDos(toDosUpdate);
    setDones([...dones, ...toAdd]);
  };

  const addItem = (ItemToAdd) => {
    const toDosCopy = [...toDos];
    toDosCopy.push(ItemToAdd);
    setToDos(toDosCopy);
  };

  return (
    <div className="App">
      <button onClick={() => getMatches()}>getMatches</button>
      <div className="content">
        <List
          title={"TO DO"}
          itemList={toDos}
          actionCheck={toDone}
          onSumbitAction={addItem}
        />
        <List title={"DONE"} itemList={dones} actionCheck={toDone} />
      </div>
    </div>
  );
}
