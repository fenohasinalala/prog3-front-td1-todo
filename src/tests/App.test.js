import renderer from "react-test-renderer";
import App from "../App";

it("renders the landing page", () => {
  const component = renderer.create(<App />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
