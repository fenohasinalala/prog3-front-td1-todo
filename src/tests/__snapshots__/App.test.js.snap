// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the landing page 1`] = `
<div
  className="App"
>
  <div
    className="content"
  >
    <div
      className="list"
    >
      <h1>
        TO DO
      </h1>
      <form
        action="submit"
        onSubmit={[Function]}
      >
        <input
          onChange={[Function]}
          placeholder="Ajouter un Item"
          type="text"
          value=""
        />
      </form>
      <ul />
    </div>
    <div
      className="list"
    >
      <h1>
        DONE
      </h1>
      <ul />
    </div>
  </div>
</div>
`;
